### PR TITLE
content plugins not able to insert custom fieldsets into core forms

### PR DESCRIPTION
**Pull request by pollen8**
This is basically the work done in this previous PR
https://github.com/joomla/joomla-cms/pull/4284
With the additions that Jean Marie asked for regarding multi lingual sites and the associations tab that is added.
The test instructions are the same as #4284

---
## Background

With the implementation of using JLayout to render parameters (https://github.com/joomla/joomla-cms/pull/2162) support for rendering custom named fieldsets ceased to be supported. This is a documented feature since at least Joomla 2.5 http://docs.joomla.org/Adding_custom_fields_to_the_article_component and is also documented to be suported in 3.1 at http://docs.joomla.org/Adding_custom_fields_to_core_components_using_a_plugin. In fact, the current core user profile plugin would break if JLayout 'joomla.edit.params' was implemented for com_users.user due to the plugin using custom named fieldsets (https://github.com/joomla/joomla-cms/blob/staging/plugins/user/profile/profiles/profile.xml#L3-L4) and the current implementation of JLayout 'joomla.edit.params' being effectively hardcoded to only render fieldsets named `params` and `attribs` (https://github.com/joomla/joomla-cms/blob/staging/layouts/joomla/edit/params.php#L15-L18). 
## Changes

This PR changes the behavior of JLayout 'joomla.edit.params' to render all fieldsets of the form, except those designated with the `ignore_fieldsets` property.

Affected by this PR are the following backend edit views:
- com_categories.category
- com_contact.contact
- com_content.article
- com_finder.filter
- com_menus.item
- com_modules.module
- com_newsfeeds.newsfeed
- com_plugins.plugin
- com_tags.tag
- com_templates.style
## Backwards Compatibility

This PR should be 100% backwards compatible while restoring this functionality.
## Testing

Apply the patch associated with this PR and review the above views to ensure that all of the currently rendered fieldsets (as tabs in Isis) remain to be rendered. Then, install and plugin that uses a custom named fieldset and ensure that it is rendered at all of the above views as described in Joomla's documentation.

A test plugin can be downloaded from https://github.com/betweenbrain/Joomla-Custom-Field-Plugin-Example/archive/test.zip

### CODE REVIEW

File: layouts/joomla/edit/params.php [hunk 1]
```diff
@@ -9,15 +9,9 @@
 
 defined('_JEXEC') or die;
 
-$app = JFactory::getApplication();
-$form = $displayData->getForm();
-
-$fieldSets = $form->getFieldsets('params');
-
-// For BC with versions < 3.2 we need to render the attribs too
-$attribsFieldSet = $form->getFieldsets('attribs');
```
Comment (zero-24) [inline layouts/joomla/edit/params.php]: hmm are you sure you can rmove that? See the comment one line above.

**Comment by wilsonge** [comment]
Thank you for doing this :)

**Comment by shre001** [comment]
@this works good<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7126">issues.joomla.org/joomla-cms/7126</a>.</sub>

**Comment by designbengel** [comment]
Applied Patch -> Tag edit (/index.php?option=com_tags&view=tag&layout=edit) wrong or missing language string COM_TAGS_URLS_FIELDSET_LABEL

Also could not test com_weblinks - sorry, where can i get it, couldn´t find at JED ?

Then i installed your test-plugin and the custom named fieldset were shown like expected.

So just one small issue with the tag view...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7126">issues.joomla.org/joomla-cms/7126</a>.</sub>

**Comment by zero-24** [comment]
>Also could not test com_weblinks - sorry, where can i get it, couldn´t find at JED ?

The JED Version is broken :smile: 

Please try the RC2 from here: https://github.com/joomla-extensions/weblinks/releases/download/3.4.1-rc2/pkg_weblinks_dev.zip until we have a final 3.4.2 of weblinks :smiley: 

Decoupled Extensions (later also e.g. messages, banners etc.) can be found here: http://extensions.joomla.org/extensions/extension/official-extensions/

File: layouts/joomla/edit/params.php (hunk 1 above)
Comment (pollen8) [inline layouts/joomla/edit/params.php]: $attribsFieldSet isn't used in the layout , so pretty sure that removing it won't change rendering

**Comment by pollen8** [comment]
@designbengel Thanks for pointing that out. That fieldset was being added twice. It should render just a hidden field at the bottom of the form, so I've added it to the ignore_fieldsets array to avoid it being rendered as a tab.

**Comment by pollen8** [comment]
@designbengel if you could retest once more then we will have 2 successful tests :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7126">issues.joomla.org/joomla-cms/7126</a>.</sub>

**Comment by designbengel** [comment]
Yes, now everything is fine :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7126">issues.joomla.org/joomla-cms/7126</a>.</sub>

**Comment by zero-24** [comment]
RTC :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7126">issues.joomla.org/joomla-cms/7126</a>.</sub>

**Comment by pollen8** [comment]
Yeah! I've got that warm Friday Joomla feeling - thanks everyone for testing

**Comment by Kubik-Rubik** [comment]
Thank you @pollen8 and @betweenbrain! Merged with https://github.com/joomla/joomla-cms/commit/580bec7336786b8d332ac08d2154b42b6ae281d8.

**Comment by infograf768** [comment]
Multilanguage. Associations on.
This has created an issue when editing a menu item.  We get 2 associations tabs.
Possible PR here:
https://github.com/joomla/joomla-cms/pull/7582

**Comment by infograf768** [comment]
#7582 is now merged
I wonder, concerning the other places where it has been added in this PR if is not missing 
`$this->fieldsets = array();` 
before using
```
// Fieldsets to not automatically render by /layouts/joomla/edit/params.php
$this->ignore_fieldsets = array('item_associations');
```
instead of placing this code on top of file without defining $this->fieldsets.

**Pull request closed without merging**